Package path resolution broken within @tamagui/vite-plugin on Windows platforms

### Current Behavior

The Vite Plugin does not work correctly on Windows, due to incorrect package path resolution. This failure prevents transitive dependencies like "react-native" and "react-native-svg" from being properly aliased to their web alternatives when using packages like "@tamagui/lucide-icons" with Vite on Windows.

For example, `resolve('@tamagui/react-native-svg')` evaluates to `C:\C:\workspace\starter-free\node_modules\@tamagui\react-native-svg\dist\esm\index.mjs`, which generates an Error like this:

```
✘ [ERROR] Could not read from file: C:\C:\workspace\starter-free\node_modules\@tamagui\react-native-svg\dist\esm\index.mjs

    node_modules/@tamagui/lucide-icons/dist/esm/icons/book-template.mjs:3:26:
      3 │ import { Path, Svg } from "react-native-svg";
```

### Expected Behavior

Instead, `resolve('@tamagui/react-native-svg')` should evaluate to `C:\workspace\starter-free\node_modules\@tamagui\react-native-svg\dist\esm\index.mjs`

### Tamagui Version

```markdown
Since v1.112.2: this commit: https://github.com/tamagui/tamagui/commit/4914410c2b04aaeb798edccfaa885135dc702d42
```

### Platform (Web, iOS, Android)

```markdown
Web (Vite)
```

### Reproduction

```markdown
npm create tamagui@latest
Ok to proceed? (y) y
√ Project name: ... bug-repro
√ Pick a template: » Learn - Vite + Webpack, Tamagui config from scratch
cd bug-repro
rm .yarnrc
echo "nodeLinker: node-modules" > .yarnrc.yml
yarn
yarn add @tamagui/lucide-icons
mv ./vite.config.ts ./vite.config.mts
patch ./src/Root.tsx

@@ -1,6 +1,7 @@
 import '@tamagui/core/reset.css'

 import { Button, TamaguiProvider, YStack } from 'tamagui'
+import { Plus } from '@tamagui/lucide-icons'
 import { LinearGradient } from 'tamagui/linear-gradient'

 import config from './tamagui.config'
@@ -9,7 +10,7 @@
   return (
     <TamaguiProvider config={config} defaultTheme="light">
       <YStack flex={1} items="center" justify="center">
-        <Button>Hello world</Button>
+        <Button icon={Plus}>Hello world</Button>
         <LinearGradient zIndex={-1} fullscreen colors={['red', 'blue']} />
       </YStack>
     </TamaguiProvider>

yarn dev:vite
```

### System Info

```markdown
System:
    OS: Windows 10 10.0.26100
    CPU: (16) x64 AMD Ryzen 7 5800X 8-Core Processor
    Memory: 7.23 GB / 31.89 GB
  Binaries:
    Node: 20.11.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 4.5.0 - C:\Program Files\nodejs\yarn.CMD
    npm: 10.5.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (132.0.2957.115)
    Internet Explorer: 11.0.26100.1882
  npmPackages:
    @pmmmwh/react-refresh-webpack-plugin: ^0.5.11 => 0.5.11
    @tamagui/animations-css: ^1.125.23 => 1.125.23
    @tamagui/config: ^1.125.23 => 1.125.23
    @tamagui/core: ^1.125.23 => 1.125.23
    @tamagui/create-theme: ^1.125.23 => 1.125.23
    @tamagui/lucide-icons: ^1.125.23 => 1.125.23
    @tamagui/react-native-svg: ^1.125.23 => 1.125.23
    @tamagui/vite-plugin: ^1.125.23 => 1.125.23
    @vitejs/plugin-react-swc: ^3.6.0 => 3.6.0
    async-retry: 1.3.3 => 1.3.3
    babel-polyfill: ^6.26.0 => 6.26.0
    css-loader: ^6.10.0 => 6.10.0
    esbuild-loader: ^4.1.0 => 4.1.0
    esbuild-register: ^3.5.0 => 3.5.0
    expo-linear-gradient: ^12.7.2 => 12.7.2
    file-loader: ^6.2.0 => 6.2.0
    html-webpack-plugin: ^5.6.0 => 5.6.0
    mini-css-extract-plugin: ^2.8.1 => 2.8.1
    react: * => 18.3.1
    react-dom: * => 18.3.1
    react-native-svg: ^15.11.2 => 15.11.2
    react-native-web: ^0.19.10 => 0.19.10
    react-refresh: ^0.14.0 => 0.14.0
    style-loader: ^3.3.4 => 3.3.4
    tamagui: ^1.125.23 => 1.125.23
    tamagui-loader: ^1.125.23 => 1.125.23
    thread-loader: ^4.0.2 => 4.0.2
    url-loader: ^4.1.1 => 4.1.1
    vite: ^6.0.6 => 6.0.6
    vite-plugin-require: ^1.2.14 => 1.2.14
    webpack: ^5.91.0 => 5.91.0
    webpack-bundle-analyzer: ^4.10.1 => 4.10.1
    webpack-cli: ^5.1.4 => 5.1.4
    webpack-dev-server: ^5.0.4 => 5.0.4
```

--- a/code/compiler/vite-plugin/src/plugin.ts
+++ b/code/compiler/vite-plugin/src/plugin.ts
@@ -13,7 +13,7 @@ const fileURLToPathName = (fileUrl?: string) => {
   }
   return pathname;
 };
-const resolve = async (name: string) => fileURLToPathName(await import.meta.resolve?.(name))
+const resolve = (name: string) => fileURLToPathName(import.meta.resolve?.(name));
 
 export function tamaguiPlugin({
   optimize,

--- a/code/compiler/vite-plugin/src/plugin.ts
+++ b/code/compiler/vite-plugin/src/plugin.ts
@@ -3,8 +3,17 @@ import type { Plugin } from 'vite'
 import { transformWithEsbuild } from 'vite'
 import { tamaguiExtractPlugin } from './extract'
 import { Static, loadTamaguiBuildConfig, tamaguiOptions } from './loadTamagui'
+import url from 'url'
 
-const resolve = (name: string) => import.meta.resolve?.(name).replace('file://', '')
+const fileURLToPathName = (fileUrl?: string) => {
+  if (!fileUrl) return undefined;
+  let { pathname } = new url.URL(fileUrl);
+  if (process.platform === 'win32') {
+    pathname = pathname.replace(/^\/([A-Za-z]:)/, '$1');
+  }
+  return pathname;
+};
+const resolve = async (name: string) => fileURLToPathName(await import.meta.resolve?.(name))
 
 export function tamaguiPlugin({
   optimize,

--- a/code/compiler/vite-plugin/src/plugin.ts
+++ b/code/compiler/vite-plugin/src/plugin.ts
@@ -3,11 +3,10 @@ import type { Plugin } from 'vite'
 import { transformWithEsbuild } from 'vite'
 import { tamaguiExtractPlugin } from './extract'
 import { Static, loadTamaguiBuildConfig, tamaguiOptions } from './loadTamagui'
-import url from 'url'
 
 const fileURLToPathName = (fileUrl?: string) => {
   if (!fileUrl) return undefined;
-  let { pathname } = new url.URL(fileUrl);
+  let { pathname } = new URL(fileUrl);
   if (process.platform === 'win32') {
     pathname = pathname.replace(/^\/([A-Za-z]:)/, '$1');
   }

--- a/code/compiler/vite-plugin/src/plugin.ts
+++ b/code/compiler/vite-plugin/src/plugin.ts
@@ -3,16 +3,13 @@ import type { Plugin } from 'vite'
 import { transformWithEsbuild } from 'vite'
 import { tamaguiExtractPlugin } from './extract'
 import { Static, loadTamaguiBuildConfig, tamaguiOptions } from './loadTamagui'
+import { fileURLToPath } from 'url'
 
-const fileURLToPathName = (fileUrl?: string) => {
-  if (!fileUrl) return undefined;
-  let { pathname } = new URL(fileUrl);
-  if (process.platform === 'win32') {
-    pathname = pathname.replace(/^\/([A-Za-z]:)/, '$1');
-  }
-  return pathname;
+const safeFileURLToPath = (fileURL?: string) => {
+  if (!fileURL) return undefined;
+  return fileURLToPath(fileURL);
 };
-const resolve = (name: string) => fileURLToPathName(import.meta.resolve?.(name));
+const resolve = (name: string) => safeFileURLToPath(import.meta.resolve?.(name));
 
 export function tamaguiPlugin({
   optimize,

